src/visitpy/common/Logging.C

```diff
@@ -1640,19 +1649,112 @@ static std::string log_ResetPickAttributesRPC(ViewerRPC *rpc)
+
+    std::string all;
+    for(size_t i = 0; i < size_t(aol->GetNumAnnotations()); ++i)
+    {
+        const AnnotationObject &annot = aol->GetAnnotation(i);
+        if(annot.GetActive())
+        {
+            std::string typeName;
+            if (annot.GetObjectType() == AnnotationObject::LegendAttributes)
+            {
+                // since legends are associated with a plot,  the object
+                // name will be something like 'Plot0000'.  So, use the
+                // object name, replacing 'Plot' with 'legend' to better
+                // better associate it with the correct plot.
+                typeName=annot.GetObjectName();
+                typeName.replace(0,4, "legend");
+            }
+            else
+            {
+                typeName=(AnnotationObject::AnnotationType_ToString(annot.GetObjectType()));
+                std::transform(typeName.begin(), typeName.end(), typeName.begin(), ::tolower);
+            }
+            std::string s = typeName +
+                            " = GetAnnotationObject(\"" +
+                            annot.GetObjectName() +
+                            "\")\n";
+            if(annot.GetVisible())
+                s += typeName + ".visible = 0\n";
+            else
+                s += typeName + ".visible = 1\n";
+            all += s + "\n";
+        }
+    }
+
+    return all;
 }
 
 static std::string log_DeleteActiveAnnotationObjectsRPC(ViewerRPC *rpc)
 {
-    return MESSAGE_COMMENT("DeleteActiveAnnotationObjects", MSG_NOT_IMPLEMENTED);
+    AnnotationObjectList *aol =
+        viewer->GetViewerState()->GetAnnotationObjectList();
+
+    std::string all;
+    for(size_t i = 0; i < size_t(aol->GetNumAnnotations()); ++i)
+    {
+        const AnnotationObject &annot = aol->GetAnnotation(i);
+        if(annot.GetActive())
+        {
+            std::string typeName;
+            if (annot.GetObjectType() == AnnotationObject::LegendAttributes)
+            {
+                // since legends are associated with a plot,  the object
+                // name will be something like 'Plot0000'.  So, use the
+                // object name, replacing 'Plot' with 'legend' to better
+                // better associate it with the correct plot.
+                typeName=annot.GetObjectName();
+                typeName.replace(0,4, "legend");
+            }
+            else
+            {
+                typeName=(AnnotationObject::AnnotationType_ToString(annot.GetObjectType()));
+                std::transform(typeName.begin(), typeName.end(), typeName.begin(), ::tolower);
+            }
+            std::string s = typeName +
+                           " = GetAnnotationObject(\"" +
+                           annot.GetObjectName() +
+                           "\")\n";
+             s += typeName + ".Delete()\n";
+            all += s + "\n";
+        }
+    }
+
+    return all;
 }
 
+// this isn't relevant in the cli.
+#if 0
```

Comment:
can we just delete the code block then?

Reply:
We could, awaiting comments on whether it would be needed in CLI or not.